hugr/src/ops/constant.rs

```diff
@@ -96,20 +129,71 @@ pub enum Value {
     },
 }
 
-/// Boxed [`CustomConst`] trait object.
+/// An opaque newtype awround a `Box<dyn CustomConst>`.
 ///
-/// Use [`Value::extension`] to create a new variant of this type.
-///
-/// This is required to avoid <https://github.com/rust-lang/rust/issues/78808> in
-/// [`Value::Extension`], while implementing a transparent encoding into a
-/// `CustomConst`.
+/// This will be the serialisation barrier that ensures all implementors of
+/// [`CustomConst`] are serialised through [`CustomSerialized`].
 #[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
 #[serde(transparent)]
-pub struct ExtensionValue(pub(super) Box<dyn CustomConst>);
+pub struct ExtensionValue {
+    #[serde(serialize_with = "ExtensionValue::serialize_with")]
+    #[serde(deserialize_with = "ExtensionValue::deserialize_with")]
+    #[serde(flatten)]
+    v: Box<dyn CustomConst>,
```

Comment:
If you flatten this field, don't you need to get rid of the `v` in
```python
class CustomConst(ConfiguredBaseModel):
    c: str
    v: Any
```
and instead accept extra args in the model?

Reply:
Ah nevermind, you set `content = "v"` in the typetag for `CustomConst` 👍